solr/solrj-zookeeper/src/test/org/apache/solr/common/cloud/SolrZkClientTest.java

```diff
@@ -281,4 +282,42 @@ public void testCheckInterrupted() {
     SolrZkClient.checkInterrupted(new InterruptedException());
     assertTrue(Thread.currentThread().isInterrupted());
   }
+
+  @Test
+  public void testWithSolrResourceLoader() {
+    setupSystemProperties();
+    SolrResourceLoader solrResourceLoader =
+        new SolrResourceLoader(Path.of("."), ClassLoader.getSystemClassLoader());
+    new SolrZkClient.Builder()
+        .withUrl(zkServer.getZkHost())
+        .withTimeout(AbstractZkTestCase.TIMEOUT, TimeUnit.MILLISECONDS)
+        .withSolrClassLoader(solrResourceLoader)
+        .build()
+        .close(); // no more tests needed. We only test class instantiation
+    clearSystemProperties();
```

Comment:
[Q] Do you know whether these system properties actually need cleared?  Solr tests all run with a JUnit '@Rule' (called "SystemPropertiesRestoreRule") that is supposed to clear them in between each test case.

If that's not working, I can file a separate ticket to fix that.  But if it is, we should remove redundant sysprop removal here.

Reply:
Thanks @gerlowskija , didn't know that _SystemPropertiesRestoreRule_ was part of _SolrTestCase_
Actually I followed the existing style, as I have seen this logic used in other classes.
Removed it.
Updated changes.txt file as well.
Thank you for your time on this